action/protocol/staking/candidate.go

```diff
@@ -94,6 +94,11 @@ func (d *Candidate) Validate() error {
 	return nil
 }
 
+// isSelfStakeBucketSettled checks if self stake bucket is settled
+func (d *Candidate) isSelfStakeBucketSettled() bool {
+	return d.SelfStakeBucketIdx != candidateNoSelfStakeBucketIndex
```

Comment:
candidateNoSelfStakeBucketIndex is a special value?

Reply:
yes, it means candidate has not already been self-staked

Reply:
rename to `_candidateNoSelfStakeBucketIndex``

Reply:
what does it mean a self stake bucket is settled?

Reply:
This method is used to wrap the code `d.SelfStakeBucketIdx != candidateNoSelfStakeBucketIndex` to avoid making this judgement in multiple places.

Basically, it returns `true` in three cases: 
- **Candidate Activated**: The candidate is in an active state and has a valid SelfStakeBucketIdx(the bucket can be self-owned or endorsed). 
- **Candidate Endorsement Expired**: The endorsement of the candidate has expired (since the expiration of endorsement takes effect with delay, the SelfStakeBucketIdx will not be cleared before performing other operations on the bucket). 
- **Candidate Unstaked**: The candidate has been unstaked (currently, the implementation of unstake does not clear the SelfStakeBucketIdx, but it might be more appropriate to clear it).